pymc/distributions/continuous.py

```diff
@@ -1019,8 +1019,41 @@ def logcdf(value, mu, lam, alpha):
             msg="mu > 0, lam > 0, alpha >= 0",
         )
 
-    def icdf(mu, lam):
-        return stats.invgauss(mu=mu, scale=lam)
+    def icdf(value, mu, lam, alpha=0):
+        """Calculate the inverse cumulative distribution function (icdf) of the Wald distribution.
+
+        Args
+        ----
+            value (float): Probability value between 0 and 1.
+            mu (float): Mean of the distribution.
+            lam (float): Scale of the distribution.
+
+        Returns
+        -------
+            float: The value x such that P(W <= x) = p, where W is the Wald distribution with given mu and lam.
+
+        Raises
+        ------
+            ValueError: If value is not between 0 and 1.
+            ValueError: If lam is not positive.
+        """
```

Comment:
Better not to include docs. Users are not supposed to use these methods directly. Instead they will call `pm.icdf(rv, value)` which will ultimately call this with the right arguments.

The docstrings are also incorrect because inputs and outputs are TensorVariables (not floats), and errors are not really raised until the expression is evaluated. These docstrings are really tricky to write, as people usually come to these with a numpy framework in mind, and they don't match the actual code. That's why we removed all of them quite recently (no docstrings is better than misleading docstrings)

Reply:
ok, I understand. It's removed now